packages/pretty-format/src/plugins/ImmutableList.js

```diff
@@ -4,28 +4,25 @@
  * This source code is licensed under the BSD-style license found in the
  * LICENSE file in the root directory of this source tree. An additional grant
  * of patent rights can be found in the PATENTS file in the same directory.
+ *
  * @flow
  */
 
 'use strict';
 
-const printImmutable = require('./printImmutable');
-const IS_LIST_SENTINEL = '@@__IMMUTABLE_LIST__@@';
+import type {Colors, Indent, Options, Print} from '../types.js';
 
-const isList = (maybeList: Object) => {
-  return !!(maybeList && maybeList[IS_LIST_SENTINEL]);
-};
+const printImmutable = require('./lib/printImmutable');
 
-const test = (object: Object) => object && isList(object);
+const IS_LIST = '@@__IMMUTABLE_LIST__@@';
+const test = (maybeList: any) => !!(maybeList && maybeList[IS_LIST]);
 
 const print = (
-  val: Object, 
-  print: Function,
-  indent: Function,
-  opts: Object,
-  colors: Object
-) => {
-  return printImmutable(val, print, indent, opts, colors, 'List', false);
-};
+  val: any,
+  print: Print,
+  indent: Indent,
+  opts: Options,
+  colors: Colors,
+) => printImmutable(val, print, indent, opts, colors, 'List', false);
 
 module.exports = {print, test};
```

Comment:
Another option is to type-check the whole thing using the `Plugin` type
`module.exports = ({print, test}: Plugin);`